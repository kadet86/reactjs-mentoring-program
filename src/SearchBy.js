import React from 'react';
import {Button} from 'primereact/button';

export default function SearchBy({searchBy, onChange}) {
    return (
        <div className="search-by">
            <label>SEARCH BY</label>
            <Button
                className={searchBy === 'title' ? 'p-button-danger' : 'p-button-secondary'}
                data-value="title" 
                onClick={onChange} 
                label="Title" />
            <Button
                className={searchBy === 'genres' ? 'p-button-danger' : 'p-button-secondary'}
                data-value="genres" 
                onClick={onChange}
                label="Genre" />
        </div>
    );
}