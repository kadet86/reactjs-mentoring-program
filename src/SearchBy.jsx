// @flow
import * as React from 'react';
import styled from 'styled-components';
import Button from './Button';

type Props = {
    searchBy: string,
    onChange: (event: SyntheticEvent<HTMLButtonElement>) => void,
};

const SearchByContainer = styled.div`
    display: inline-block;
    font-size: 12px;

    label {
        margin-right: 20px;
        color: white;
    }

    button {
        margin-right: 12px;
        font-size: 12px;
    }
`;

export default function SearchBy({ searchBy, onChange }: Props) {
    return (
        <SearchByContainer>
            <label>SEARCH BY</label>
            <Button
                type="button"
                className={
                    searchBy === 'title' ? 'button-danger' : 'button-secondary'
                }
                data-value="title"
                onClick={onChange}
                label="Title"
            />
            <Button
                type="button"
                className={
                    searchBy === 'genres' ? 'button-danger' : 'button-secondary'
                }
                data-value="genres"
                onClick={onChange}
                label="Genre"
            />
        </SearchByContainer>
    );
}
