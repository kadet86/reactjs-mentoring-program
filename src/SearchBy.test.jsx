import React from 'react';
import {shallow} from 'enzyme';
import SearchBy from './SearchBy';

describe('SearchBy component', () => {
    it('has persistent default snapshot', () => {
        const component = shallow(<SearchBy />);
        expect(component).toMatchSnapshot();
    });

    it('has persistent snapshot with searchBy="title"', () => {
        const component = shallow(<SearchBy searchBy="title"/>);
        expect(component).toMatchSnapshot();
    });

    it('has persistent snapshot with searchBy="genres"', () => {
        const component = shallow(<SearchBy searchBy="genres"/>);
        expect(component).toMatchSnapshot();
    });
});
