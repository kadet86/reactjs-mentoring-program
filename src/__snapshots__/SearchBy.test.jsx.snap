// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBy component has persistent default snapshot 1`] = `ShallowWrapper {}`;

exports[`SearchBy component has persistent snapshot with searchBy="genres" 1`] = `ShallowWrapper {}`;

exports[`SearchBy component has persistent snapshot with searchBy="title" 1`] = `ShallowWrapper {}`;
